crates/optimism/txpool/src/maintain.rs

```diff
@@ -75,3 +105,103 @@ where
         }
     }
 }
+
+/// Returns a spawnable future for maintaining the state of the transaction pool.
+pub fn maintain_transaction_pool_interop_future<N, Pool, St>(
+    pool: Pool,
+    events: St,
+    supervisor_client: SupervisorClient,
+) -> BoxFuture<'static, ()>
+where
+    N: NodePrimitives,
+    Pool: TransactionPool + 'static,
+    Pool::Transaction: MaybeInteropTransaction,
+    St: Stream<Item = CanonStateNotification<N>> + Send + Unpin + 'static,
+{
+    async move {
+        maintain_transaction_pool_interop(pool, events, supervisor_client).await;
+    }
+    .boxed()
+}
+
+/// Maintains the state of the transaction pool by handling new blocks and reorgs.
+///
+/// This listens for any new blocks and reorgs and updates the transaction pool's state accordingly
+pub async fn maintain_transaction_pool_interop<N, Pool, St>(
+    pool: Pool,
+    mut events: St,
+    supervisor_client: SupervisorClient,
+) where
+    N: NodePrimitives,
+    Pool: TransactionPool,
+    Pool::Transaction: MaybeInteropTransaction,
+    St: Stream<Item = CanonStateNotification<N>> + Send + Unpin + 'static,
+{
+    let metrics = MaintainPoolInteropMetrics::default();
+    let supervisor_client = Arc::new(supervisor_client);
+    loop {
+        let Some(event) = events.next().await else { break };
+        if let CanonStateNotification::Commit { new } = event {
+            if new.is_empty() {
+                continue;
+            }
```

Comment:
we can remove this check here because for op the new block is never empty

Reply:
done for interop and cond